tests/jecoute/FileStatsServiceTest.php

```diff
@@ -0,0 +1,154 @@
+                ],
+                2 => [
+                    'uuid' => '',
+                    'content' => 'Est-ce une question à choix unique ?',
+                    'type' => 'unique_choice',
+                    'stats' => [
+                        'answers' => [
+                            0 => [
+                                'value' => 'Réponse A',
+                                'percent' => '50.00',
+                                'answered' => '1',
+                            ],
+                            1 => [
+                                'value' => 'Réponse B',
+                                'percent' => '50.00',
+                                'answered' => '1',
+                            ],
+                        ],
+                    ],
+                    'surveyQuestion' => $surveyQuestion,
+                ],
+                3 => [
+                    'uuid' => '',
+                    'content' => 'Ceci est-il un champ libre d\'une question suggérée ?',
+                    'type' => 'simple_field',
+                    'stats' => [],
+                    'surveyQuestion' => $surveyQuestion,
+                ],
+            ],
+        ];
+
+        $this->statisticsProvider->expects($this->once())->method('getStatsBySurvey')->willReturn($findDataByQuestionReturn);
+        $this->dataAnswerRepository
+            ->expects($this->exactly(2))
+            ->method('findAllBySurveyQuestion')
+            ->willReturn([
+                ['textField' => 'Test Unit', 'postedAt' => new \DateTime('now')],
+                ['textField' => 'Test Unit 2', 'postedAt' => '2019-01-01 01:00:00'],
+            ], [])
+        ;
+
+        $statFile = $this->fileStatsService->getStatsFileCsvBySurvey($survey);
+
+        $this->assertArrayHasKey('filename', $statFile);
+        $this->assertArrayHasKey('content', $statFile);
+        $this->assertInternalType('string', $statFile['content']);
+
+        $this->assertSame(1, substr_count($statFile['content'], 'Questionnaire TestU '));
+        $this->assertSame(1, substr_count($statFile['content'], 'Aucune donnée n\'est disponible pour le moment.'.\PHP_EOL));
+        $this->assertSame(1, substr_count($statFile['content'], 'Est-ce une question à choix multiple ?'.\PHP_EOL));
+        $this->assertSame(1, substr_count($statFile['content'], 'Réponse A;66.67%;2'.\PHP_EOL));
+        $this->assertSame(1, substr_count($statFile['content'], 'Réponse A;50.00%;1'.\PHP_EOL));
+        $this->assertSame(1, substr_count($statFile['content'], 'Test Unit 2;2019-01-01 01:00:00'.\PHP_EOL));
+    }
+
+    public function testGetFilenameCsv(): void
+    {
+        $survey = new Survey();
+        $survey->setName('test u');
+        $response = $this->invokeMethod($this->fileStatsService, 'getFilenameCsv', [$survey]);
+        $this->assertSame('test_u_'.date('Y-m-d_H-i').'.csv', $response);
+    }
+
+    private function invokeMethod(
```

Comment:
IMHO, we are not supposed to test private methods.
We only test the API of a class, aka the public methods.

No matter how the private methods behaves: only the API (public methods) will have an impact on the outside (the application)

So.. i think you can remove this hack and tests about privates methods, what do you think ?

Reply:
ok removed